sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -250,11 +256,46 @@ export class MessageSender extends LinkEntity {
    * @param sendBatch Boolean indicating whether the encoded message represents a batch of messages or not
    * @return {Promise<Delivery>} Promise<Delivery>
    */
-  private _trySend(encodedMessage: Buffer, sendBatch?: boolean): Promise<void> {
-    const retryOptions = this._getSenderOptions?.retryOptions || {};
-    retryOptions.timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);
+  private _trySend(
+    encodedMessage: Buffer,
+    sendBatch: boolean,
+    options?: OperationOptions
+  ): Promise<void> {
+    const abortSignal = options?.abortSignal;
+    const retryTimeoutInMs =
+      this._getSenderOptions?.retryOptions?.timeoutInMs || Constants.defaultOperationTimeoutInMs;
     const sendEventPromise = () =>
       new Promise<void>(async (resolve, reject) => {
+        const rejectOnAbort = () => {
+          const desc: string =
+            `[${this._context.namespace.connectionId}] The send operation on the Sender "${this.name}" with ` +
+            `address "${this.address}" has been cancelled by the user.`;
+          // Cancellation is user-intended, so log to info instead of warning.
+          log.error(desc);
+          return reject(new AbortError("The send operation has been cancelled by the user."));
+        };
+
+        if (abortSignal && abortSignal.aborted) {
+          // operation has been cancelled, so exit quickly
+          return rejectOnAbort();
+        }
+
+        const removeListeners = (): void => {
+          clearTimeout(waitTimer);
+          if (abortSignal) {
+            abortSignal.removeEventListener("abort", onAborted);
```

Comment:
abortSignal?.removeEventListener() and remove the if()?

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/8159#issuecomment-608254608